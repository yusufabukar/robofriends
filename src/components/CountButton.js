import React, { Component } from 'react';

class CountButton extends Component {
    constructor() {
        super();
        this.state = {
            count: 0
        };
    };

    shouldComponentUpdate(nextProps, nextState) {
        if (this.state.count !== nextState.count) {
            return true;
        } else {
            return false;
        };
    };

    updateCount = () => {
        this.setState(state => {
            return {count: state.count + 1};
        });
    };

    render() {
        return (
            <button onClick={this.updateCount}>
                Count: {this.state.count}
            </button>
        );
    };
};

export default CountButton;