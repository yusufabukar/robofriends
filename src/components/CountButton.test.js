import React from 'react';
import { shallow } from 'enzyme';
import CountButton from './CountButton';

it('Renders Count Button component', () => {
    const mockColour = 'red';
    expect(shallow(<CountButton colour={mockColour} />)).toMatchSnapshot();
});

it('Correctly increments counter', () => {
    const mockColour = 'green';
    const wrapper = shallow(<CountButton colour={mockColour} />);

    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 2});
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 3});
    wrapper.find('[id="counter"]').simulate('keypress');
    expect(wrapper.state()).toEqual({count: 3});
    expect(wrapper.props().colour).toEqual('green');
});