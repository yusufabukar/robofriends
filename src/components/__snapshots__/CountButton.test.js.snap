// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Count Button component 1`] = `
<button
  colour="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
